src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -441,6 +444,14 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     fun addSuperinterface(superinterface: KClass<*>)
         = addSuperinterface(superinterface.asTypeName())
 
+    fun addSuperinterface(superinterface: ParameterSpec) = apply {
```

Comment:
Yeah, I see your point. I don't feel good about implicitly modifying the primary constructor though, at the very least it's not intuitively clear that `superinterfaceBy()` would do this, and we'll have to resolve duplicate `ParamSpec`s coming from this method and the primary constructor's `FunSpec`. In my opinion, it's safer to leave primary constructor creation unchanged, and add a way to map the name of a specific primary constructor param to a superinterface we're adding. But I might be missing some edge cases here.

@JakeWharton @swankjesse what are your thoughts on the API?